### PR TITLE
Help refactoring

### PR DESCRIPTION
**Pull request by mcquin**
More help refactoring:

* Create `cellprofiler.gui.help` package
* Convert help menu content to reST and move to `cellprofiler.gui.help.content` (`wx`-free)
  * Updated outdated links.
* Create `cellprofier.gui.menu.Menu` base class for menus
* Extract `CPFrame` help menu to `cellprofiler.gui.help.menu.Menu` and help menu searching to `cellprofiler.gui.help.search.Search`
  * Move search functionality from `cellprofiler.gui.manual` to `cellprofier.gui.help.search`
  * Delete `cellprofiler.gui.manual`
* Create `cellprofiler.gui.welcome` for the CellProfiler welcome screen
  * Move welcome screen pages from `cellprofiler.gui.html.content` to `cellprofiler.gui.welcome`
  * Extract welcome screen link following from `cellprofiler.gui.html.htmlwindow.HtmlClickablWindow` to `cellprofiler.gui.welcome.Content`

I don't entirely love the package structure right now but these changes enable us to include help content from the GUI without `wx`, reduce duplication, remove the unused `cellprofiler.gui.manual` module, etc.

After this, I think we can tackle #2840.

### CODE REVIEW

**Comment by mcquin** [comment]
Previous Help menu:
screen shot 2017-09-03 at 2 08 33 pm

New Help menu:
screen shot 2017-09-03 at 2 10 03 pm

Differences:
* Removed "Online Manual": we don't have one that's up to date.
* "Data Tool Help" is redundant with "Data Tools > Help".  I updated "Help > Navigating The Menu Bar > Using the Data Tools Menu" with instructions on where to find the data tools help.
* "Module help": Do we really need 3 different ways to view the current module's help?

File: cellprofiler/gui/help/menu.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
+import cellprofiler.gui.help.content
+import cellprofiler.gui.help.search
+import cellprofiler.gui.htmldialog
+import cellprofiler.gui.menu
+import cellprofiler.modules
+
+
+class Menu(cellprofiler.gui.menu.Menu):
```
Comment (0x00b1) [inline cellprofiler/gui/help/menu.py]: 👍

**Comment by 0x00b1** [comment]
I'm getting errors when searching. For example, when I search for `foo`:

```sh
wm0eb-41b:CellProfiler agoodman$ cellprofiler
<string>:118: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:36: (ERROR/3) Unknown target name: "below".
<string>:2: (ERROR/3) Unknown target name: "metadata".
<string>:28: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:38: (ERROR/3) Unexpected indentation.
<string>:51: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:6: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:69: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:9: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:4: (ERROR/3) Unexpected indentation.
<string>:6: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:35: (ERROR/3) Unknown target name: "below".
<string>:6: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:33: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:36: (ERROR/3) Unexpected indentation.
<string>:49: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:56: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:23: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:26: (ERROR/3) Unexpected indentation.
<string>:7: (WARNING/2) Definition list ends without a blank line; unexpected unindent.
<string>:10: (ERROR/3) Unexpected indentation.
<string>:14: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:15: (WARNING/2) Definition list ends without a blank line; unexpected unindent.
<string>:10: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:4: (ERROR/3) Unexpected indentation.
<string>:5: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:5: (ERROR/3) Unexpected indentation.
<string>:9: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:3: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:2: (WARNING/2) Inline emphasis start-string without end-string.
<string>:3: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:10: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:13: (ERROR/3) Unexpected indentation.
<string>:20: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:21: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:4: (ERROR/3) Unexpected indentation.
<string>:5: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:16: (WARNING/2) Bullet list ends without a blank line; unexpected unindent.
<string>:4: (ERROR/3) Unexpected indentation.
<string>:7: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:23: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:25: (ERROR/3) Unexpected indentation.
<string>:35: (ERROR/3) Unknown target name: "below".
<string>:3: (WARNING/2) Block quote ends without a blank line; unexpected unindent.
<string>:3: (WARNING/2) Title underline too short.

MeasureColocalization
=====
```

**Comment by mcquin** [comment]
@0x00b1 Those are, unfortunately, expected until all help documentation (including module settings) are converted properly to reST.

**Comment by AnneCarpenter** [comment]
FWIW, regarding this Q:  
_Module help": Do we really need 3 different ways to view the current module's help?_
I think it might be disability-friendly to have it in the menu vs. as buttons, but I'm not an expert about that.

**Pull request merged**